app/services/katello/ui_notifications/subscriptions/expire_soon.rb

```diff
@@ -0,0 +1,56 @@
+module Katello
+  module UINotifications
+    module Subscriptions
+      class ExpireSoon
+        class << self
+          def deliver!
+            subs_expiring_soon.each do |subscription|
+              next if notification_already_exists?(subscription)
+              ::Notification.create!(
+                :subject => subscription,
+                :initiator => User.anonymous_admin,
+                :audience => Notification::AUDIENCE_ADMIN,
+                :message => ::UINotifications::StringParser.new(
+                  blueprint.message,
+                  :subject => subscription
+                ),
+                :actions => {
+                  :links => [
+                    {
+                      # Can't use simply a :subscription_path on the blueprint
+                      # since this is a Bastion page
+                      :href => "/subscriptions/#{subscription.id}",
+                      :title => _('Subscription'),
+                      :external => true
+                    }
+                  ]
+                },
+                :notification_blueprint => blueprint
+              )
+            end
+          end
+
+          def subs_expiring_soon
+            result = []
+            Katello::Subscription.unscoped.all.each do |subscription|
```

Comment:
is there no better query than go over all? at least I would use select instead of the next statement below.